guides/cookbook.md

```diff
@@ -579,6 +780,78 @@ examining the type annotations of each top-level struct and comparing against
 not match `"foo"`, the reader can quickly skip to the start of the next value.
 
 <script>writeTabs()</script>
+<div class="tabpane C" markdown="1">
+```c
+#include <cstdlib>
+#include "ionc/ion.h"
+
+#define ION_OK(x) if (x) { printf("Error: %s\n", ion_error_to_str(x)); return x; }
+
+void ion_string_from_cstr(const char *cstr, ION_STRING *out) {
+    out->value = (BYTE *)cstr;
+    out->length = (SIZE)strlen(cstr);
+}
+
+int main(int argc, char **argv) {
+    const char ion_binary[] = { ... };    // bytes representing the Ion binary encoding of the above data
+    const int ion_binary_length = ...;    // the number of bytes in ion_binary
+
+    hREADER reader;
+    ION_READER_OPTIONS options;
+
+    memset(&options, 0, sizeof(ION_READER_OPTIONS));
+
+    ION_OK(ion_reader_open_buffer(&reader,
+                                  (BYTE *)ion_binary,
+                                  ion_binary_length,
+                                  &options));
+
+    ION_STRING foo;
+    ion_string_from_cstr("foo", &foo);
+
+    int sum = 0;
+
+    ION_TYPE ion_type;
+    ION_OK(ion_reader_next(reader, &ion_type));
+
+    while (ion_type != tid_EOF) {
+        if (ion_type == tid_STRUCT) {
+            BOOL annotation_found;
+            ION_OK(ion_reader_has_annotation(reader, &foo, &annotation_found));
+            if (annotation_found) {
+                ION_OK(ion_reader_step_in(reader));
+
+                ION_TYPE ion_type2;
+                ION_OK(ion_reader_next(reader, &ion_type2));
+
+                while (ion_type2 != tid_EOF) {
+                    ION_STRING ion_string;
+                    ION_OK(ion_reader_get_field_name(reader, &ion_string));
+                    char *field_name = ion_string_strdup(&ion_string);
+                    if (strcmp(field_name, "quantity") == 0) {
```

Comment:
I recommend creating an ION_STRING for "quantity" and use `ION_STRING_EQUALS` (from `ion_string.h`) instead of allocating a new `char *` and using `strcmp` for each comparison.

Reply:
+1. Also `strcmp` has some issues and we should be using `strncmp` if we really have to 
https://www.geeksforgeeks.org/difference-strncmp-strcmp-c-cpp/